### PR TITLE
feat: Add metainfo in datablock

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

File: src/query/datablocks/src/meta_info.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::sync::Arc;
+
+#[typetag::serde(tag = "type")]
+pub trait MetaInfo: Send + Sync {
```
Comment (Xuanwo) [inline src/query/datablocks/src/meta_info.rs:19]: Why we need to introduce a trait for this?

Comment (zhyass) [reply src/query/datablocks/src/meta_info.rs:19]: > Why we need to introduce a trait for this?

The output and input of the pipeline rely on the datablock. by adding metainfo to the datablock, some additional information can be passed downstream. This is very useful in delete, update, etc.

**Review by zhang2014 — APPROVED** [review]

File: src/query/datablocks/src/meta_info.rs (hunk 1 above)
Comment (BohuTANG) [reply src/query/datablocks/src/meta_info.rs:19]: It would be better to add a unit test to show how to use it.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
cc @youngsofun

**Pull request merged**